entity-service/src/main/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationRepository.kt

```diff
@@ -0,0 +1,67 @@
+package com.egm.stellio.entity.authorization
+
+import com.egm.stellio.entity.model.Property
+import org.neo4j.ogm.session.Session
+import org.springframework.stereotype.Component
+
+@Component
+class Neo4jAuthorizationRepository(
+    private val session: Session
+) {
+
+    fun filterEntitiesUserHasOneOfGivenRights(
+        userId: String,
+        entitiesId: List<String>,
+        rights: Set<String>
+    ): List<String> {
+        val query = """
+            MATCH (userEntity:Entity {id : ${'$'}userId}), (entity:Entity)
+            WHERE entity.id IN ${'$'}entitiesId
+            MATCH (userEntity)-[:HAS_OBJECT]->(right:Attribute:Relationship)-[]->(entity:Entity)
+            WHERE size([label IN labels(right) WHERE label IN ${'$'}rights]) > 0
+            return entity.id as id
+            UNION
+            MATCH (userEntity)-[:HAS_OBJECT]->(:Attribute:Relationship)-[:IS_MEMBER_OF]->(:Entity)-[:HAS_OBJECT]-(grpRight:Attribute:Relationship)-[]->(entity:Entity)
+            WHERE size([label IN labels(grpRight) WHERE label IN ${'$'}rights]) > 0
+            return entity.id as id
+        """.trimIndent()
+
+        val parameters = mapOf(
+            "userId" to userId,
+            "entitiesId" to entitiesId,
+            "rights" to rights
+        )
+
+        return session.query(query, parameters).map {
+            it["id"] as String
+        }
+    }
+
+    fun getUserRoles(userId: String): List<String> {
+        val query = """
+            MATCH (userEntity:Entity { id: ${'$'}userId })
+            OPTIONAL MATCH  (userEntity)-[:HAS_VALUE]->(p:Property { name: "https://ontology.eglobalmark.com/authorization#roles" })
```

Comment:
Maybe you can declare and use EGM_ROLES constant (as you did for tests)

Reply:
Yes, indeed